src/imputation/tmi_imputation.py

```diff
@@ -0,0 +1,337 @@
+import pandas as pd
+import math
+import numpy as np
+from src.staging.pg_conversion import sic_to_pg_mapper
+
+def apply_to_original(filtered_df, original_df):
+    original_df.loc[filtered_df.index] = filtered_df
+    return original_df
+
+
+def filter_by_column_content(
+    raw_df: pd.DataFrame, column: str, column_content: list
+) -> pd.DataFrame:
+
+    clean_df = raw_df[raw_df[column].isin(column_content)].copy()
+
+    return clean_df
+
+
+def impute_pg_by_sic(df: pd.DataFrame, sic_mapper: pd.DataFrame):
+    """Impute missing product groups for companies that do no r&d
+    and status = "Form sent out" using the SIC number ('rusic').
+
+    Args:
+        df (pd.DataFrame): initial dataframe from staging
+        sic_mapper (pd.DataFrame): SIC to PG mapper
+
+    Returns:
+        pd.DataFrame: Returns the original dataframe with new PGs
+        overwriting the missing values
+    """
+    
+    df['200'] = df['200'].astype("category")
+    
+    long_df = filter_by_column_content(df, "formtype", ["0001"])
+
+    # Filter for q604 = No or status = "Form sent out"
+    filtered_data = long_df.loc[
+        (long_df["status"] == "Form sent out") | (long_df["604"] == "No")
```

Comment:
Could simplify by using just [] as in: filtered_data = long_df[{long_df["status"] == "Form sent out") | (long_df["604"] == "No")]